system-test/vision.js

```diff
@@ -0,0 +1,359 @@
+      vision.detectLabels([
+        IMAGES.logo,
+        IMAGES.rushmore
+      ], function(err, labels) {
+        assert.ifError(err);
+
+        assert.strictEqual(labels.length, 2);
+
+        assert(labels[0].indexOf('logo') > -1);
+        assert(labels[1].indexOf('classical sculpture') > -1);
+
+        done();
+      });
+    });
+
+    it('should support verbose mode', function(done) {
+      var options = {
+        verbose: true
+      };
+
+      vision.detectLabels(IMAGES.rushmore, options, function(err, labels) {
+        assert.ifError(err);
+
+        assert(is.defined(labels[0].mid));
+
+        done();
+      });
+    });
+  });
+
+  describe('landmarks', function() {
+    it('should detect landmarks from an image', function(done) {
+      vision.detectLandmarks(IMAGES.rushmore, function(err, landmarks) {
+        assert.ifError(err);
+
+        assert.deepEqual(landmarks, ['Mount Rushmore']);
+
+        done();
+      });
+    });
+
+    it('should detect landmarks from multiple images', function(done) {
+      vision.detectLandmarks([
+        IMAGES.logo,
+        IMAGES.rushmore
+      ], function(err, landmarks) {
+        assert.ifError(err);
+
+        assert.strictEqual(landmarks.length, 2);
+        assert.deepEqual(landmarks[0], []);
+        assert.deepEqual(landmarks[1], ['Mount Rushmore']);
+
+        done();
+      });
+    });
+
+    it('should support verbose mode', function(done) {
+      var opts = {
+        verbose: true
+      };
+
+      vision.detectLandmarks(IMAGES.rushmore, opts, function(err, landmarks) {
+        assert.ifError(err);
+
+        assert(is.defined(landmarks[0].mid));
+
+        done();
+      });
+    });
+  });
+
+  describe('logos', function() {
+    it('should detect logos from an image', function(done) {
+      vision.detectLogos(IMAGES.logo, function(err, logos) {
+        assert.ifError(err);
+
+        assert.deepEqual(logos, ['Google']);
+
+        done();
+      });
+    });
+
+    it('should detect logos from multiple images', function(done) {
+      vision.detectLogos([
+        IMAGES.rushmore,
+        IMAGES.logo
+      ], function(err, logos) {
+        assert.ifError(err);
+
+        assert.strictEqual(logos.length, 2);
+        assert.deepEqual(logos[0], []);
+        assert.deepEqual(logos[1], ['Google']);
+
+        done();
+      });
+    });
+
+    it('should support verbose mode', function(done) {
+      var options = {
+        verbose: true
+      };
+
+      vision.detectLogos(IMAGES.logo, options, function(err, logos) {
+        assert.ifError(err);
+
+        assert(is.defined(logos[0].mid));
+
+        done();
+      });
+    });
+  });
+
+  describe('properties', function() {
+    it('should detect image properties', function(done) {
+      vision.detectProperties(IMAGES.rushmore, function(err, properties) {
+        assert.ifError(err);
+
+        assert.deepEqual(properties.colors, [
+          '#3b3027',
```

Comment:
Oh weird, I thought I caught the tests.

On Mon, Mar 7, 2016, 3:51 PM Dave Gramlich <notifications@github.com> wrote:

> In system-test/vision.js
> <https://github.com/GoogleCloudPlatform/gcloud-node/pull/1136#discussion_r55269809>
> :
>
> > +        assert.ifError(err);
> > +
> > +        assert(is.defined(logos[0].mid));
> > +
> > +        done();
> > +      });
> > +    });
> > +  });
> > +
> > +  describe('properties', function() {
> > +    it('should detect image properties', function(done) {
> > +      vision.detectProperties(IMAGES.rushmore, function(err, properties) {
> > +        assert.ifError(err);
> > +
> > +        assert.deepEqual(properties.colors, [
> > +          '#3b3027',
>
> It looks like rgb-hex does not include the # in the return values, so
> this test fails now. :(
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-node/pull/1136/files#r55269809>
> .
>